apps/settings/js/panels/operator_settings/views/network_type_selector.js

```diff
@@ -0,0 +1,88 @@
+define(function(require) {
+  'use strict';
+
+  var DialogService = require('modules/dialog_service');
+  var Module = require('modules/base/module');
+  var CustomizedNetworkTypeMap = require('modules/customized_network_type_map');
+  var ListView = require('modules/mvvm/list_view');
+  var NetworkTypeItemTemplate =
+    require('panels/operator_settings/views/network_type_item_template');
+
+  var NetworkTypeSelector = Module.create(function NetworkTypeSelector(root) {
+    this._root = root;
+    this._manager = null;
+    this._customizedNetworkTypeMap = null;
+    this._networkTypeMap = null;
+
+    var networkTypeItemTemplate =
+      NetworkTypeItemTemplate(this._getTextInfoForNetworkType.bind(this));
+
+    this._networkTypeOptions =
+      ListView(this._root, null, networkTypeItemTemplate);
+    
+    this._root.addEventListener('blur', () => {
+      this._manager.setPreferredNetworkType(this._root.value).catch(() => {
+        DialogService.alert('preferredNetworkTypeAlertErrorMessage', {
+          title: 'preferredNetworkTypeAlertTitle'
+        });
+      });
+    });
+    this._boundUpdate = this._update.bind(this);
+  });
+
+  NetworkTypeSelector.prototype._getTextInfoForNetworkType = function(type) {
+    if (this._customizedNetworkTypeMap &&
+        type in this._customizedNetworkTypeMap) {
+      return {
+        text: this._customizedNetworkTypeMap[type]
+      };
+    } else if (this._networkTypeMap) {
+      return {
+        l10nId: this._networkTypeMap(type)
+      };
+    } else {
+      return type;
+    }
+  };
+
+  NetworkTypeSelector.prototype._update = function(type) {
```

Comment:
`update` is too vague to understand, can we add the noun for it to make it more specific ?